### ISSUE TITLE
"Error: Cannot use JSX unless the '--jsx' flag is provided" when using "extends"

### ISSUE THREAD

**Issue by elado** [atom/ide-typescript#52]
My `tsconfig.json` has `compilerOptions.jsx="react"` that comes from an `"extends"` property. I'm seeing this error.

```json
{
  "extends": "./node_modules/[buildsystempackage]/tsconfig.json",
   ...
}
```

**Comment by miketierney** [comment]
Is there any update on this? I'm having the same issue.

**Comment by atdrago** [comment]
I have the same issue. We have multiple projects sharing the same `tsconfig.json` via an npm module in our org. Our local `tsconfig.json` looks like this:
```
{
    "extends": "./node_modules/@org/project-name/tsconfig-react"
}
```
And the `tsconfig-react` file referenced above looks like this:
```
{
    "compilerOptions": {
        "jsx": "preserve"
    },
    "extends": "./tsconfig-base.json"
}
```
If I move `"compilerOptions": { "jsx": "preserve" }` into our local config, I no longer see this error.

**Comment by jsrj** [comment]
Has this issue been resolved yet? I have not been able to find a definitive answer on how to fix this. I have been encountering this error "out of the box" while trying to start working with .TSX files. Attempted the above suggestions with no success.

**Comment by BiosBoy** [comment]
Faced with this issue too. The solution for local tsconfig.json: 
```
    "compilerOptions": {
        "jsx": "preserve"
    },
```
does not help in my case :(...

VS Code version: 1.25.1
Type-Script version: 3.0.1
Aawesome-typescript-loader": 5.2.0

My tsconfig.base.json:
```
{
  "compilerOptions": {
    "allowJs": true,
    "checkJs": false,
    "module": "commonjs",
    "target": "es5",
    "jsx": "preserve",
    "moduleResolution": "node",
    "lib": ["es2015", "dom"]
  }
}
```

**Comment by kduffey-lyft** [comment]
Can you try:
    "compilerOptions": {
      "jsx": "react"
    }

See if that helps. For me, within Webstorm, that seems to solve the problem. Though, jsx: preserve also did so.  Another comment I found elsewhere indicated using 

"include": [
  "./src/ts/**/*"
]

in the config as well.  Give those a try.. see if either/both solve it.

**Comment by kevinduffey** [comment]
Same issue within Webstorm. From elsewhere, I discovered adding:

    "compilerOptions": {
      "jsx": "react"
    },
    "include": [
      "./src/ts/**/*"
    ]

to my local tsconfig did the trick. For me, without the include, it still worked, but added that just in case. For me, "jsx": "preseve" actually worked as well. 

Hopefully that helps.

**Comment by damieng** [comment]
Should now be fixed with the new language server - try ide-typescript 0.8.0

**Comment by pk-developer** [comment]
Restart your IDE. Sometimes tsconfig.json changes aren't immediately picked up

**Comment by andrinurwakhid** [comment]
check on your tsconfig.json add     "jsx": "preserve"
this works for me

**Comment by kakadshivaji70** [comment]
Thanks for the solution @atdrago

**Comment by Goblin80** [comment]
Every time I run `npm start`, it overrides whatever I configure in `{jsx: ...}` with `react-jsx` in order to be compatible with JSX transform in React 17.

```bash
The following changes are being made to your tsconfig.json file:
  - compilerOptions.jsx must be react-jsx (to support the new JSX transform in React 17)
```

The problem is VSCode using an older version of typescript (4.0.3), while the typescript version shipped with the project is (4.1.2).

The following did the trick for me:

1. Go to the command palette <kbd>CTRL</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd>.
2. Choose "TypeScript: Select a TypeScript Version...".
2. Choose "Use workspace Version".

**Comment by gugol2** [comment]
> Every time I run `npm start`, it overrides whatever I configure in `{jsx: ...}` with `react-jsx` in order to be compatible with JSX transform in React 17.
> 
> ```shell
> The following changes are being made to your tsconfig.json file:
>   - compilerOptions.jsx must be react-jsx (to support the new JSX transform in React 17)
> ```
> 
> The problem is VSCode using an older version of typescript (4.0.3), while the typescript version shipped with the project is (4.1.2).
> 
> The following did the trick for me:
> 
> 1. Go to the command palette CTRL+Shift+P.
> 2. Choose "TypeScript: Select a TypeScript Version...".
> 3. Choose "Use workspace Version".

In my case both my project and VS Code were using the same 'problematic' version of TS (4.0.3) which is what CRA version "4.0.1" uses when you create a project (I am under node v10.19.0).

So I had to do both things:
1. Upgrade TS from version 4.0.3 to 4.1.2
2. Choose "Use workspace Version" (4.1.2) as you said.

Thanks!

**Comment by designervoid** [comment]
> Every time I run `npm start`, it overrides whatever I configure in `{jsx: ...}` with `react-jsx` in order to be compatible with JSX transform in React 17.
> 
> ```shell
> The following changes are being made to your tsconfig.json file:
>   - compilerOptions.jsx must be react-jsx (to support the new JSX transform in React 17)
> ```
> 
> The problem is VSCode using an older version of typescript (4.0.3), while the typescript version shipped with the project is (4.1.2).
> 
> The following did the trick for me:
> 
> 1. Go to the command palette CTRL+Shift+P.
> 2. Choose "TypeScript: Select a TypeScript Version...".
> 3. Choose "Use workspace Version".

Also if u cant find this option in command pallete, you should install typescript globally:
```
npm install -g typescript
```
